beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -41,25 +44,215 @@ func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (
+	exits := vs.ExitPool.PendingExits(head, req.Slot, false /*noLimit*/)
+	validExits := make([]*ethpb.SignedVoluntaryExit, 0, len(exits))
+	for _, exit := range exits {
+		val, err := head.ValidatorAtIndexReadOnly(exit.Exit.ValidatorIndex)
+		if err != nil {
+			log.WithError(err).Warn("Proposer: invalid exit")
+			continue
+		}
+		if err := blocks2.VerifyExitAndSignature(val, head.Slot(), head.Fork(), exit, head.GenesisValidatorsRoot()); err != nil {
+			log.WithError(err).Warn("Proposer: invalid exit")
+			continue
+		}
+		validExits = append(validExits, exit)
+	}
+	blk.Body().SetVoluntaryExits(validExits)
+
+	// Set sync aggregate. New in Altair
+	if slots.ToEpoch(req.Slot) >= params.BeaconConfig().AltairForkEpoch {
+		syncAggregate, err := vs.getSyncAggregate(ctx, req.Slot-1, bytesutil.ToBytes32(parentRoot))
+		if err != nil {
+			return nil, errors.Wrap(err, "could not compute the sync aggregate")
+		}
+		if err := blk.Body().SetSyncAggregate(syncAggregate); err != nil {
+			return nil, errors.Wrap(err, "could not set sync aggregate")
+		}
+	}
+
+	// Set execution data. New in Bellatrix
+	if slots.ToEpoch(req.Slot) >= params.BeaconConfig().BellatrixForkEpoch {
+		fallBackToLocal := true
+		canUseBuilder, err := vs.canUseBuilder(ctx, req.Slot, idx)
+		if err != nil {
+			log.WithError(err).Warn("Proposer: failed to check if builder can be used")
+		}
+		if canUseBuilder && err != nil {
+			h, err := vs.getPayloadHeaderFromBuilder(ctx, req.Slot, idx)
+			if err != nil {
+				log.WithError(err).Warn("Proposer: failed to get payload header from builder")
+			} else {
+				blk.SetBlinded(true)
+				if err := blk.Body().SetExecution(h); err != nil {
+					log.WithError(err).Warn("Proposer: failed to set execution payload")
+				} else {
+					fallBackToLocal = false
+				}
+			}
+		}
+		if fallBackToLocal {
+			executionData, err := vs.getExecutionPayload(ctx, req.Slot, idx, bytesutil.ToBytes32(parentRoot), head)
+			if err != nil {
+				return nil, errors.Wrap(err, "could not get execution payload")
+			}
+			if err := blk.Body().SetExecution(executionData); err != nil {
+				return nil, errors.Wrap(err, "could not set execution payload")
+			}
+		}
+	}
+
+	// Set bls to execution change. New in Capella
+	if slots.ToEpoch(req.Slot) >= params.BeaconConfig().CapellaForkEpoch {
+		changes, err := vs.BLSChangesPool.BLSToExecChangesForInclusion(head)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not pack BLSToExecutionChanges")
+		}
+		if err := blk.Body().SetBLSToExecutionChanges(changes); err != nil {
+			return nil, errors.Wrap(err, "could not set BLSToExecutionChanges")
+		}
+	}
+
+	if err := sBlk.SetBlock(blk); err != nil {
 		return nil, err
 	}
-	return vs.getBellatrixBeaconBlock(ctx, req)
+	sr, err := vs.computeStateRoot(ctx, sBlk)
+	if err != nil {
+		return nil, fmt.Errorf("could not compute state root: %v", err)
+	}
+	blk.SetStateRoot(sr)
+
+	pb, err := blk.Proto()
+	if err != nil {
+		return nil, err
+	}
+	switch {
```

Comment:
Is there any reason to use a switch statement here instead of a sequence of if checks? Also I would start with Capella 
```go
if slots.ToEpoch(req.Slot) >= params.BeaconConfig().CapellaForkEpoch {
    return &ethpb.GenericBeaconBlock{Block: &ethpb.GenericBeaconBlock_Capella{Capella: pb.(*ethpb.BeaconBlockCapella)}}, nil
} 
if slots.ToEpoch(req.Slot) >= params.BeaconConfig().BellatrixForkEpoch {    
...
```